### PR TITLE
Fix minor issues with build command

### PR DESCRIPTION
**Pull request by NatashaWso2**
- write the balx to the target path given by user if he/she has provided one
- fix issue when providing the source/package to be built with a "/" at the end

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/BuilderUtils.java [hunk 1]
```diff
@@ -64,7 +64,7 @@ public static void compileWithTestsAndWrite(Path sourceRootPath,
 
         if (skiptests) {
             outStream.println();
-            compiler.write(bLangPackage, packagePath);
+            compiler.write(bLangPackage, targetPath);
```
Comment (Kishanthan) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/BuilderUtils.java]: Why we have changed this to targetPath?

Comment (NatashaWso2) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/BuilderUtils.java]: We should write the balx to the target path given by the user. If the target path is not given then by default we write the balx inside the target folder.

**Comment by NatashaWso2** [comment]
Added a test case with 9ae847c

**Comment by NatashaWso2** [comment]
Changed the commit message to be have 50 chars

File: tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/PackagingInitTestCase.java [hunk 2]
```diff
@@ -316,6 +316,40 @@ public void testInitWithInvalidPackage() throws Exception {
...
+        Assert.assertTrue(Files.exists(sourceFilePath));
+        Assert.assertTrue(Files.exists(projectPath.resolve("Ballerina.toml")));
+
+        // Remove the .ballerina inside the project because we only want the bal file
+        Files.deleteIfExists(projectPath.resolve(".ballerina").resolve(".gitignore"));
+        Files.deleteIfExists(projectPath.resolve(".ballerina"));
+
+        // Test ballerina build
+        String[] clientArgsForBuild = {"main.bal", "-o", genExecPath.resolve("foo.bal").toString()};
+        ballerinaServer = createNewBallerinaServer();
```
Comment (Kishanthan) [inline tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/PackagingInitTestCase.java]: Do we have to create a new server instance here?

Comment (NatashaWso2) [reply tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/PackagingInitTestCase.java]: yes the server instance created is deleted after executing a command as in https://github.com/ballerina-platform/ballerina-lang/blob/master/tests/ballerina-integration-test-utils/src/main/java/org/ballerinalang/test/context/ServerInstance.java#L345

Comment (Kishanthan) [reply tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/PackagingInitTestCase.java]: I think we have to revisit how we have written tools integration test cases. We have followed the approach of unzip only once and run all the test cases for the ballerina integration test module. We have to follow the same here in order to reduce time with each test cases.

File: tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/PackagingInitTestCase.java [hunk 3]
```diff
@@ -316,6 +316,40 @@ public void testInitWithInvalidPackage() throws Exception {
         runMainFunction(projectPath, generatedBalx.toString());
     }
 
+    @Test(description = "Test building a project with a single bal file using the target path")
+    public void testBuildWithTarget() throws Exception {
```
Comment (Kishanthan) [inline tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/PackagingInitTestCase.java]: Is there two different test scenarios covered from this test method? If yes, we have to move them to two different methods.

Comment (NatashaWso2) [reply tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/PackagingInitTestCase.java]: No only one test scenario is tested if the target file is created at the correct place.

**Comment by NatashaWso2** [comment]
@Kishanthan rebased the PR with master and updated

**Pull request merged**